pallets/parachain-staking/src/lib.rs

```diff
@@ -1462,6 +1529,148 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Check if an account has been migrated from lock to freeze
+		/// If not migrated, perform the migration
+		/// Returns true if migration was performed, false if already migrated
+		///
+		/// `is_collator` determines whether the account is a collator or delegator
+		fn check_and_migrate_lock(
+			account: &T::AccountId,
+			is_collator: bool,
+		) -> Result<bool, DispatchError> {
+			use frame_support::traits::{fungible::MutateFreeze, LockableCurrency};
+
+			// Check if already migrated
+			if is_collator {
+				if <MigratedCandidates<T>>::contains_key(account) {
+					return Ok(false); // Already migrated
+				}
+			} else {
+				if <MigratedDelegators<T>>::contains_key(account) {
+					return Ok(false); // Already migrated
+				}
+			}
+
+			// Not migrated yet, proceed with migration
+			let (lock_id, freeze_reason) = if is_collator {
+				(COLLATOR_LOCK_ID, FreezeReason::StakingCollator)
+			} else {
+				(DELEGATOR_LOCK_ID, FreezeReason::StakingDelegator)
+			};
+
+			// Get the amount that should be locked/frozen from storage
+			let amount = if is_collator {
+				// For collators, get the bond amount from storage
+				<CandidateInfo<T>>::get(account)
+					.map(|info| info.bond)
+					.unwrap_or_default()
```

Comment:
As migrate_locks_to_freezes_batch is permisionless, anyone can put dummy account and spam the storage items MigratedCandidates and MigratedDelegators. We must add a check that ensure the collator/delegator exist.
We should probably replace return an Option here and handle separatly the cases "exist with zero amount" and "not exist"